Codec/Archive/Tar/Index/Utils.hs

```diff
@@ -63,10 +65,16 @@ readInt32BE :: BS.ByteString -> Int -> Int32
 readInt32BE bs i = fromIntegral (readWord32BE bs i)
 {-# INLINE readInt32BE #-}
 
+readWord32OffPtrBE :: Ptr Word32 -> Int -> IO Word32
+readWord32OffPtrBE ptr i = do
+  case targetByteOrder of
```

Comment:
`targetByteOrder` was broken before GHC 8.10, so it would be more robust to check `#if defined(WORDS_BIGENDIAN)`.